src/avram/pool_statement_logging.cr

```diff
@@ -5,39 +5,31 @@ module DB
       statement_with_retry &.exec
     end
 
-    def exec(*args) : ExecResult
-      log_query(args)
-      statement_with_retry &.exec(*args)
-    end
-
-    def exec(args : Array) : ExecResult
-      log_query(args)
-      statement_with_retry &.exec(args)
+    def exec(*args_, args : Array? = nil) : ExecResult
+      log_query(*args_, args: args)
+      statement_with_retry &.exec(*args_, args: args)
     end
 
     def query : ResultSet
       log_query
       statement_with_retry &.query
     end
 
-    def query(*args) : ResultSet
-      log_query(args)
-      statement_with_retry &.query(*args)
-    end
-
-    def query(args : Array) : ResultSet
-      log_query(args)
-      statement_with_retry &.query(args)
+    def query(*args_, args : Array? = nil) : ResultSet
+      log_query(*args_, args: args)
+      statement_with_retry &.query(*args_, args: args)
     end
 
-    def scalar(*args)
-      log_query(args)
-      statement_with_retry &.scalar(*args)
+    def scalar(*args_, args : Array? = nil)
+      log_query(*args_, args: args)
+      statement_with_retry &.scalar(*args_, args: args)
     end
 
-    private def log_query(args = [] of String)
+    private def log_query(*args_, args : Array? = nil)
       Avram.settings.query_log_level.try do |level|
-        Avram.logger.log(level, {query: @query, args: args})
+        logging_args = EnumerableConcat.build(args_, args)
```

Comment:
Where does `EnumerableConcat` come from? I don't see any docs on it, and it doesn't come up [in search](https://github.com/crystal-lang/crystal/search?q=EnumerableConcat&unscoped_q=EnumerableConcat)

Reply:
It's a helper class using in crystal-db to implement the `*args/args:Array`. I actually didn't expect to expose it so it's a `:nodoc:` https://github.com/crystal-lang/crystal-db/blob/master/src/db/enumerable_concat.cr .

The whole idea is to ensure a single Enumerable for arguments without allocating extra instances unless needed. And is due to the lack of logging that it made it's way up to avram.

Reply:
ah, cool. Thanks!